compiler/codegen.js

```diff
@@ -3533,6 +3585,34 @@ const generateUpdate = (scope, decl, _global, _name, valueUnused = false) => {
 const generateIf = (scope, decl) => {
   const out = truthy(scope, generate(scope, decl.test), getNodeType(scope, decl.test), false, true);
 
+  // todo: support ||, not just fastOr
```

Comment:
ngl I don't really like all this added complexity just for some built-in cases in `if`s

Reply:
Should I just optimize ifs that have one type check, or remove this completely?

Reply:
I think for now this can just be removed

Reply:
Done!

Reply:
A lot of builtins are still unneededly changed for this?

Reply:
I will say especially once #116 is merged those are really moreso optimizations, so we don't introduce an unnecessary local. But, I'll revert those changes regardless

Reply:
I think it is nicer code generally fwiw, we should have more general optimizations later which handles this sort of stuff rather than focusing on a few select scenarios imo